src/credentials/CHIPCertToX509.cpp

```diff
@@ -801,15 +792,9 @@ CHIP_ERROR DecodeConvertTBSCert(TLVReader & reader, ASN1Writer & writer, ChipCer
         err = DecodeConvertSubjectPublicKeyInfo(reader, writer, certData);
         SuccessOrExit(err);
 
-        // If the next element is a certificate extension...
-        err = reader.Next();
+        // certificate extensions
+        err = DecodeConvertExtensions(reader, writer, certData);
```

Comment:
I may be mistaken but should not we peacefully ignore a TLV as unknown in case we couldn't decode it?

Reply:
Good Point! 
There is another extension type "FutureExtension" https://github.com/project-chip/connectedhomeip/issues/4316 - I plan to work on adding support for future extensions immediately after this PR. In any other case the function should return error and fail.

Reply:
hmm... you mean the intention is to record but ignore an error returned by this function?

Reply:
According to the spec only 6 extensions are supported by CHIP Certificates:
```
extension => CHOICE OF
{
  basic-cnstr [1] : basic-constraints,
  key-usage [2] : UNSIGNED INTEGER [range 16bits],
  extended-key-usage [3] : ARRAY [length 1..] OF key-purpose-id,
  subject-key-id [4] : OCTET STRING [length 20],
  authority-key-id [5] : OCTET STRING [length 20],
  future-extension [6] : OCTET STRING,
}
```

Any other TLV tag, not specified by the spec, SHALL result in decoding error and certificate decoding should fail.

On the other hand, the purpose of future-extension is to support any arbitrary DER encoded extensions, which won't be interpreted by CHIP SDK 1.0 but the encoding/decoding should work fine for them.

Reply:
I couldn't find a statement in the specs that calls on ignoring unknown TLVs, although I think I saw it there; will keep looking.  Anyway, not doing that doesn't seem very wise from backward compatibility point of view.

Reply:
1. Typically for future compatibility devices first receive firmware update before they can process new TLV formats.  
2. In case of certificates ignoring unknown TLV elements can introduce security breach
3. Technically it is impossible to support unknown TLV elements in certificates. If such element exists we will never be able to reproduce certificate signature and won't be able to do certificate validation.

Reply:
> Technically it is impossible to support unknown TLV elements in certificates. If such element exists we will never be able to reproduce certificate signature and won't be able to do certificate validation.

agree